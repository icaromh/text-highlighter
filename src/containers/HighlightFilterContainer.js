import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import {
  RED,
  YELLOW,
  GREEN,
} from '../constants/colors'

import { toggleFilterHighlights } from '../actions'
import CheckButton from '../components/Button/CheckButton'

const HighlightFilter = ({ handleCheckClick }) => (
  <div className="HighlightFilterContainer">
    <CheckButton onClick={handleCheckClick} value={RED} />
    <CheckButton onClick={handleCheckClick} value={YELLOW} />
    <CheckButton onClick={handleCheckClick} value={GREEN} />
  </div>
)

HighlightFilter.propTypes = {
  handleCheckClick: PropTypes.func.isRequired,
}

export {
  HighlightFilter,
}

export default connect(
  null,
  {
    handleCheckClick: toggleFilterHighlights,
  },
)(HighlightFilter)
