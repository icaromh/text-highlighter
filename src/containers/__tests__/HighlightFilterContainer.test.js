import React from 'react'
import { shallow } from 'enzyme'

import { HighlightFilter } from '../HighlightFilterContainer'

describe('#HighlightFilterContainer', () => {
  test('expect to render', () => {
    const onClickMock = jest.fn()
    const wrapper = shallow(
      <HighlightFilter handleCheckClick={onClickMock} />,
    )

    wrapper.find('CheckButton').at(0).simulate('click')

    expect(wrapper).toMatchSnapshot()
    expect(onClickMock.mock.calls.length).toBe(1)
  })
})
