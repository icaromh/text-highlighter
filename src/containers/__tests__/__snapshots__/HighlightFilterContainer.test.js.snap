// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#HighlightFilterContainer expect to render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HighlightFilter
    handleCheckClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CheckButton
          onClick={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          value="#ff8a80"
        />,
        <CheckButton
          onClick={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          value="#fff176"
        />,
        <CheckButton
          onClick={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          value="#4db6ac"
        />,
      ],
      "className": "HighlightFilterContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onClick": [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
          "value": "#ff8a80",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onClick": [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
          "value": "#fff176",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onClick": [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
          "value": "#4db6ac",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CheckButton
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
            value="#ff8a80"
          />,
          <CheckButton
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
            value="#fff176"
          />,
          <CheckButton
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
            value="#4db6ac"
          />,
        ],
        "className": "HighlightFilterContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClick": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            },
            "value": "#ff8a80",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClick": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            },
            "value": "#fff176",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClick": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            },
            "value": "#4db6ac",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
